src/Analysis/pipeline/parameter_types.py

```diff
@@ -0,0 +1,323 @@
+import abc
+
+class BaseData(metaclass=abc.ABCMeta):
+    """
+    Abbstract class od port types, that define user for
+    validation and translation to string
+    """
+    def __init__(self):
+        self.name = ""
+        """Display name of port"""
+        self.description = ""
+        """Display description of port"""
+    
+    @abc.abstractmethod 
+    def to_string(self, value):
```

Comment:
radeji  _serialize nebo _serialize_to_json 
Je to interni zalezitost. Co je parametr 'value'?